File: LaundryMgmt.Mobile/Models/CatalogModels.cs
namespace LaundryMgmt.Mobile.Models;

public enum PricingType
{
    PerItem = 0,
    WeightBased = 1
}

public enum MembershipTier
{
    None = 0,
    Silver = 1,
    Gold = 2
}

public record PaginatedList<T>(
    List<T> Items, int PageNumber, int PageSize, int TotalPages, int TotalCount, bool HasNextPage, bool HasPreviousPage);

public enum CustomerStatus
{
    Active = 0,
    Inactive = 1
}

public record CustomerListItem(
    Guid Id, string FullName, string PhoneNumber, string? Email, string? WhatsAppNumber,
    MembershipTier MembershipTier, decimal WalletBalance, int LoyaltyPoints, CustomerStatus Status);

public record CreateCustomerRequest(string FullName, string PhoneNumber, string? Email, decimal CreditLimit, string? Notes);

public record UpdateCustomerRequest(
    string FullName, string PhoneNumber, string? Email, decimal CreditLimit, MembershipTier MembershipTier, string? Notes);

public record UpdateMyProfileRequest(string FullName, string? Email, string? WhatsAppNumber);

public record CustomerAddressDto(
    Guid Id, string Label, string Line1, string? Line2, string City, string State, string PostalCode, bool IsDefault);

public record CreateCustomerAddressRequest(
    string Label, string Line1, string? Line2, string City, string State, string PostalCode, bool IsDefault);

public record CustomerDetailDto(
    Guid Id, string FullName, string PhoneNumber, string? Email,
    decimal WalletBalance, decimal CreditLimit, int LoyaltyPoints,
    MembershipTier MembershipTier, CustomerStatus Status, string? Notes, List<CustomerAddressDto> Addresses);

public record ServiceCategoryDto(Guid Id, string Name, string? Description, string? IconOrImageUrl, int DisplayOrder, bool IsActive);

public record GarmentListItem(Guid Id, string Name, Guid CategoryId, string CategoryName, string? ImageUrl);

public record CreateGarmentRequest(string Name, Guid CategoryId, string? SpecialInstructions);

public record UpdateGarmentRequest(string Name, Guid CategoryId, string? SpecialInstructions, string? ImageUrl);

public record ServiceListItem(
    Guid Id, string Name, decimal BasePrice, int EstimatedTimeHours, decimal GstPercentage, int Priority,
    string? ImageUrl, decimal ExpressSurcharge, int ExpressEtaHours);

public record ServiceDetailDto(
    Guid Id, string Name, Guid CategoryId, string CategoryName, decimal BasePrice, int EstimatedTimeHours,
    decimal GstPercentage, int Priority, string? ImageUrl, decimal ExpressSurcharge, int ExpressEtaHours);

public record CreateServiceRequest(
    string Name, Guid CategoryId, decimal BasePrice, int EstimatedTimeHours, decimal GstPercentage, int Priority,
    string? ImageUrl = null, decimal ExpressSurcharge = 0, int ExpressEtaHours = 24);

public record UpdateServiceRequest(
    string Name, Guid CategoryId, decimal BasePrice, int EstimatedTimeHours, decimal GstPercentage, int Priority,
    string? ImageUrl, decimal ExpressSurcharge, int ExpressEtaHours);

public record GarmentServicePriceDto(Guid ServiceId, string ServiceName, PricingType PricingType, decimal Price);

public record GarmentDetailDto(
    Guid Id, string Name, Guid CategoryId, string CategoryName, string? SpecialInstructions, string? ImageUrl,
    List<GarmentServicePriceDto> ServicePrices);

public record PricingMatrixServiceDto(Guid Id, string Name, Guid CategoryId, string CategoryName);

public record PricingMatrixCellDto(Guid ServiceId, PricingType? PricingType, decimal? Price);

public record PricingMatrixGarmentRowDto(Guid GarmentId, string GarmentName, Guid CategoryId, string CategoryName, List<PricingMatrixCellDto> Prices);

public record PricingMatrixDto(List<PricingMatrixServiceDto> Services, List<PricingMatrixGarmentRowDto> Garments);

File: client-web/src/app/core/models/catalog.models.ts
export enum PricingType {
  PerItem = 0,
  WeightBased = 1
}

export interface ServiceCategory {
  id: string;
  name: string;
  description?: string;
  iconOrImageUrl?: string;
  displayOrder: number;
  isActive: boolean;
}

export interface CreateServiceCategoryRequest {
  name: string;
  description?: string;
  iconOrImageUrl?: string;
  displayOrder: number;
}

export interface UpdateServiceCategoryRequest extends CreateServiceCategoryRequest {
  isActive: boolean;
}

export interface AddOn {
  id: string;
  name: string;
  description?: string;
  price: number;
  isActive: boolean;
}

export interface CreateAddOnRequest {
  name: string;
  description?: string;
  price: number;
}

export interface UpdateAddOnRequest extends CreateAddOnRequest {
  isActive: boolean;
}

export interface GarmentListItem {
  id: string;
  name: string;
  categoryId: string;
  categoryName: string;
  imageUrl?: string;
}

export interface GarmentServicePriceItem {
  serviceId: string;
  serviceName: string;
  pricingType: PricingType;
  price: number;
}

export interface GarmentDetail {
  id: string;
  name: string;
  categoryId: string;
  categoryName: string;
  specialInstructions?: string;
  imageUrl?: string;
  servicePrices: GarmentServicePriceItem[];
}

export interface CreateGarmentRequest {
  name: string;
  categoryId: string;
  specialInstructions?: string;
  imageUrl?: string;
}

export type UpdateGarmentRequest = CreateGarmentRequest;

export interface ServiceListItem {
  id: string;
  name: string;
  categoryId: string;
  categoryName: string;
  basePrice: number;
  estimatedTimeHours: number;
  gstPercentage: number;
  priority: number;
  imageUrl?: string;
  expressSurcharge: number;
  expressEtaHours: number;
}

export interface ServiceDetail {
  id: string;
  name: string;
  categoryId: string;
  categoryName: string;
  basePrice: number;
  estimatedTimeHours: number;
  gstPercentage: number;
  priority: number;
  imageUrl?: string;
  expressSurcharge: number;
  expressEtaHours: number;
}

export interface CreateServiceRequest {
  name: string;
  categoryId: string;
  basePrice: number;
  estimatedTimeHours: number;
  gstPercentage: number;
  priority: number;
  imageUrl?: string;
  expressSurcharge: number;
  expressEtaHours: number;
}

export type UpdateServiceRequest = CreateServiceRequest;

export interface PricingMatrixService {
  id: string;
  name: string;
  categoryId: string;
  categoryName: string;
}

export interface PricingMatrixCell {
  serviceId: string;
  pricingType: PricingType | null;
  price: number | null;
  expressPrice: number | null;
  gstPercentage: number | null;
  estimatedTimeHours: number | null;
  expressEtaHours: number | null;
  isActive: boolean;
}

export interface PricingMatrixGarmentRow {
  garmentId: string;
  garmentName: string;
  categoryId: string;
  categoryName: string;
  prices: PricingMatrixCell[];
}

export interface PricingMatrix {
  services: PricingMatrixService[];
  garments: PricingMatrixGarmentRow[];
}

export interface SetGarmentServicePriceRequest {
  pricingType: PricingType;
  price: number;
  expressPrice?: number | null;
  gstPercentage?: number | null;
  estimatedTimeHours?: number | null;
  expressEtaHours?: number | null;
  isActive: boolean;
}
